### ISSUE TITLE
Why no model.eval() in test code

### ISSUE THREAD

**Issue by yftongbupt** [HyoungsungKim/Relation_network_for_term_project#1]
I want to ask that why didn't you use model.eval() when you test the model?
and I found that in the code of relation network ,that code has the same problem with your code.
so I want to ask is model.eval() not necessary or what?

**Comment by HyoungsungKim** [comment]
Thank you for making a issue and visiting my repository.

Yes exactly you are right and i agree it is so weird. However while i reading few-show learning papers, i found `transduction`.

In this [paper](https://arxiv.org/pdf/1905.01436.pdf), they said that relation network(RN) is using batch normalization(BN) for transduction. I think it means that RN is using BN while testing phase.

You may know that different between model.train() and model.eval() is that model.train() uses BN but model.eval() doest not.

And somthing interesting i knew while reading few show learning papers is that all of their experiments are little different.

For example, they use different number of images when they train model.

**Comment by yftongbupt** [comment]
Thank you for your reply!

 



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Hyoungsung Kim"<notifications@github.com&gt;; 
发送时间: 2020年11月9日(星期一) 晚上6:01
收件人: "HyoungsungKim/Relation_network_for_term_project"<Relation_network_for_term_project@noreply.github.com&gt;; 
抄送: "轮廓"<597534353@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [HyoungsungKim/Relation_network_for_term_project] Why no model.eval() in test code (#1)





 
Thank you for making a issue and visiting my repository.
 
Yes exactly you are right and i agree it is so weird. However while i reading few-show learning papers, i found transduction,
 
In this paper, they said that relation network(RN) is using batch normalization(BN) for transduction. I think it means that RN is using BN while testing phase.
 
You may know that different between model.test() and model.eval() is that model.test() uses BN but model.eval() doest not.
 
And somthing interesting i knew while reading few show learning papers is that all of their experiments are little different.
 
For example, they use different number of images when they train model.
